packages/frontend/src/features/sqlRunner/components/CartesianChartConfiguration.tsx

```diff
@@ -1,14 +1,33 @@
+import { ChartKind } from '@lightdash/common';
 import { ActionIcon, Divider, Group, Stack, Title } from '@mantine/core';
 import { IconChevronDown, IconChevronRight } from '@tabler/icons-react';
 import { useState } from 'react';
 import MantineIcon from '../../../components/common/MantineIcon';
-import { BarChartFieldConfiguration } from './BarChartFieldConfiguration';
-import { BarChartStyling } from './BarChartStyling';
+import { barChartConfigSlice } from '../store/barChartSlice';
+import { useAppSelector } from '../store/hooks';
+import { lineChartConfigSlice } from '../store/lineChartSlice';
+import { CartesianChartFieldConfiguration } from './CartesianChartFieldConfiguration';
+import { LineChartStyling } from './CartesianChartStyling';
 
-export const BarChartConfig = () => {
+export const CartesianChartConfig = () => {
     const [isFieldConfigurationOpen, setIsFieldConfigurationOpen] =
         useState(true);
     const [isStylingOpen, setIsStylingOpen] = useState(false);
+    const selectedChartType = useAppSelector(
+        (state) => state.sqlRunner.selectedChartType,
+    );
+
+    const actions =
+        selectedChartType === ChartKind.LINE
+            ? lineChartConfigSlice.actions
+            : selectedChartType === ChartKind.VERTICAL_BAR
+            ? barChartConfigSlice.actions
+            : null;
```

Comment:
I think we can merge these 2
so inside `useAppSelector` we can do this logic and return actions depending on the selectedChartType since we don't use `selectedChartType` anywhere else

So like

```
const actions = useAppSelector(.... all the logic)
```

what do you think?

Reply:
Sorry, I'm not following. The actions don't get selected from the store, so they aren't in the selector. But maybe I'm missing it.

Reply:
I guess this implementation suggestion is not that important since it caused confusion. since we rely on the store state we could do: 

```
const actions = useAppSelector(state => {
     switch (selectedChartType)
          case ChartKind.LINE
            return lineChartConfigSlice.actions
....
})
```

Reply:
I think it's probably a reasonable policy that calls to useAppSelector should always return something selected from the store. What do you think?